pkg/cloudproviders/aws/cred_manager_impl.go

```diff
@@ -0,0 +1,114 @@
+package aws
+
+import (
+	"os"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/pkg/secretinformer"
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stackrox/rox/pkg/utils"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	cloudCredentialsKey = "credentials"
+)
+
+type awsCredentialsManagerImpl struct {
+	informer         *secretinformer.SecretInformer
+	secretFound      bool
+	mirroredFileName string
+	mutex            sync.RWMutex
+}
+
+var _ CredentialsManager = &awsCredentialsManagerImpl{}
+
+// NewCredentialsManager creates a new AWS credential manager.
+func NewCredentialsManager(
+	k8sClient kubernetes.Interface,
+	namespace string,
+	secretName string,
+	mirroredFileName string,
+) *awsCredentialsManagerImpl {
+	mgr := &awsCredentialsManagerImpl{mirroredFileName: mirroredFileName}
+	mgr.informer = secretinformer.NewSecretInformer(
+		namespace,
+		secretName,
+		k8sClient,
+		mgr.updateSecret,
+		mgr.updateSecret,
+		mgr.deleteSecret,
+	)
+	return mgr
+}
+
+func (c *awsCredentialsManagerImpl) updateSecret(secret *v1.Secret) {
+	if stsConfig, ok := secret.Data[cloudCredentialsKey]; ok {
+		c.mutex.Lock()
+		defer c.mutex.Unlock()
+		if err := mirrorToLocalFile([]byte(stsConfig), c.mirroredFileName); err != nil {
```

Comment:
In the case the STS config is empty (i.e. key exists in the secret, but empty contents), do we still attempt to write _and_ say `secretFound`? Seems like we should handle this case explicitly.